File: DSA/VedantChouhan_590024563/Day-71/q1_leetcode.c
#include <stdio.h>

int findJudge(int n, int trust[][2], int trustSize) {
    int score[n + 1];

    for (int i = 1; i <= n; i++) {
        score[i] = 0;
    }

    for (int i = 0; i < trustSize; i++) {
        int a = trust[i][0];
        int b = trust[i][1];

        score[a]--;
        score[b]++;
    }

    for (int i = 1; i <= n; i++) {
        if (score[i] == n - 1) {
            return i;
        }
    }

    return -1;
}

int main() {
    int n = 3;

    int trust[][2] = {
        {1, 3},
        {2, 3}
    };

    int trustSize = 2;

    printf("%d\n", findJudge(n, trust, trustSize));

    return 0;
}

File: DSA/Navneetpant_590011569/Day_71/Q1_leetcode.java
class Solution {
    public int findJudge(int n, int[][] trust) {

        int[] indegree = new int[n + 1];
        int[] outdegree = new int[n + 1];

        for (int[] edge : trust) {
            int a = edge[0];
            int b = edge[1];

            outdegree[a]++;
            indegree[b]++;
        }

        for (int i = 1; i <= n; i++) {
            if (indegree[i] == n - 1 && outdegree[i] == 0) {
                return i;
            }
        }

        return -1;
    }
}